nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -161,49 +178,18 @@ CollisionMonitor::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
 
 void CollisionMonitor::odomCallback(nav_msgs::msg::Odometry::ConstSharedPtr msg)
 {
-  // Obtaining odom_frame_id_ from first message and setting it as fixed frame for data sources
-  if (!first_odom_) {
-    for (std::shared_ptr<SourceBase> source : sources_) {
-      source->setFixedFrameId(msg->header.frame_id);
-    }
-    first_odom_ = true;
-  }
-
-  velocity_.x = msg->twist.twist.linear.x;
-  velocity_.y = msg->twist.twist.linear.y;
-  velocity_.tw = msg->twist.twist.angular.z;
-  velocity_stamp_ = msg->header.stamp;
-  velocity_received_ = true;
-}
-
-bool CollisionMonitor::velocityValid(const rclcpp::Time & curr_time)
-{
-  // Check whether velocity is not received yet
-  if (!velocity_received_) {
-    return false;
-  }
-
-  // Check that latest received odom timestamp is earlier
-  // than current time more than data_timeout_ interval
-  const rclcpp::Duration dt = curr_time - velocity_stamp_;
-  if (dt > data_timeout_) {
-    RCLCPP_WARN(
-      get_logger(),
-      "Latest odom and current collision monitor node timestamps differ on %f seconds.",
-      dt.seconds());
-    return false;
+  // Obtaining odom_frame_id_ from first message and setting it as global frame for data sources
+  if (odom_frame_id_.length() == 0) {
```

Comment:
I don't understand, where's the velocity coming from then?

Also, you should not subscribe to a topic just to get a frame ID, just make that a parameter if we no longer require the robot's current speed

Reply:
The odometry velocity usage was switched to `cmd_vel_in` to fix "twitching" problem (https://github.com/ros-planning/navigation2/pull/2982#discussion_r899039767). The time before a collision to be estimated by using input cmd_vel, which is not depending on actual robot state and thus has no flickering and positive feedback causing robot move-stop-move twitching. Since we are using it for motion estimation, I think it should be reasonable.

Now odometry is used only for getting `odom_frame_id`. I will remove this subscriber in favor of getting `odom_frame_id` from ROS-params.

Reply:
Now, `odom_fram_id` is being [obtained](https://github.com/ros-planning/navigation2/pull/2982/files#diff-e1c76d88179c14a1cf399e94bebd4f5d17dedca39e09b3bbdc810aa11a705543R207) from parameters. Odometry subscriber was completely removed.